internal/create/wizard.go

```diff
@@ -0,0 +1,310 @@
+
+// Actions returns a list of actions that can be performed on the wizard. The actions directly
+// interact with the underlying storage.
+func (w *Wizard) Actions(s *root.Store, cb ActionCallback) cui.Actions {
+	sort.Slice(w.Templates, func(i, j int) bool {
+		return w.Templates[i].Priority < w.Templates[j].Priority
+	})
+	acts := make(cui.Actions, 0, len(w.Templates))
+	for _, tpl := range w.Templates {
+		acts = append(acts, cui.Action{
+			Name: tpl.Name,
+			Fn:   mkActFunc(tpl, s, cb),
+		})
+	}
+	return acts
+}
+
+func mkActFunc(tpl Template, s *root.Store, cb ActionCallback) func(context.Context, *cli.Context) error {
+	debug.Log("creating action func for %+v, cb: %p", tpl, cb)
+	return func(ctx context.Context, c *cli.Context) error {
+		name := c.Args().First()
+		store := c.String("store")
+		force := c.Bool("force")
+
+		sec := secrets.New()
+
+		out.Print(ctx, tpl.Welcome)
+
+		// genPW is needed for the callback
+		var genPw bool
+		// password is needed for the callback
+		var password string
+		// hostname is needed in later iterations (e.g. password rule lookup)
+		var hostname string
+		// wantForName is a list of attributes that will be used to build the name
+		wantForName := set.Map(tpl.NameFrom)
+		// nameParts are the components the name will be built from
+		var nameParts []string
+		// step is only used for printing the progress
+		var step int
+		for k, v := range tpl.Attributes {
+			step++
+
+			// if no prompt is set default to the key
+			if v.Prompt == "" {
+				v.Prompt = strings.ToTitle(k)
+			}
+
+			switch v.Type {
+			case "string":
+				sv, err := termio.AskForString(ctx, fmtfn(2, strconv.Itoa(step), v.Prompt), "")
+				if err != nil {
+					return err
+				}
+				if wantForName[k] {
+					nameParts = append(nameParts, sv)
+				}
+				sec.Set(k, sv)
+			case "hostname":
+				sv, err := termio.AskForString(ctx, fmtfn(2, strconv.Itoa(step), v.Prompt), "")
+				if err != nil {
+					return err
+				}
+				hostname = extractHostname(sv)
+				if hostname == "" {
+					return fmt.Errorf("can not parse URL %s", sv)
+				}
+				if wantForName[k] {
+					nameParts = append(nameParts, hostname)
+				}
+				if u := pwrules.LookupChangeURL(hostname); u != "" {
+					sec.Set("password-change-url", u)
+				}
+				sec.Set(k, sv)
+			case "password":
+				var err error
+				genPw, err = termio.AskForBool(ctx, fmtfn(2, strconv.Itoa(step), "Generate Password?"), true)
+				if err != nil {
+					return err
+				}
+
+				if genPw {
+					password, err = generatePassword(ctx, hostname, v.Charset)
+					if err != nil {
+						return err
+					}
+				} else {
+					password, err = termio.AskForPassword(ctx, v.Prompt, true)
+					if err != nil {
+						return err
+					}
+				}
+
+				sec.SetPassword(password)
+			}
+		}
+
+		// select store
+		if store == "" {
+			store = cui.AskForStore(ctx, s)
+		}
+
+		// generate name, ask for override if already taken
```

Comment:
I don't understand that comment, where does it ask for override?

Reply:
Clarified.